src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random rand = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int number = rand.nextInt(5);
```

Comment:
Consider replacing the magic number `5` with a constant, for example, `FIGURE_COUNT`, to improve code readability and maintainability as per the checklist recommendation.